### ISSUE TITLE
Named modifiers don't work with optional arguments

### ISSUE THREAD

**Issue by onufriievkyrylo** [Vincit/objection.js#1293]
I want to provide some additional argument in named modifier, but when I'm calling it this way
```js
query.midify('filters', [filter_query])
```
it's try to find named modifier with [filter_query].
Is it expected behavior?

**Comment by koskimas** [comment]
Cannot reproduce based on your example. Works on my tests.

**Comment by onufriievkyrylo** [comment]
@koskimas Maybe my example wasn't so clear.
My model file:
```js
class Person extends Model {
  modifiers: {
    filters: (query, context) => query.where({ age: context.age })
  }
}
```
calling this modifier
```js
Person.query()
  .modify('filters', { age: 21 })
```
I'm expecting to have { age: 21 } in my modifier second argument (context). But instead I doesn't have second argument at all (undefined);

**Comment by koskimas** [comment]
Oh, ok. Yeah, that's a known limitation with the modifiers. Fixing this is a breaking change and needs to wait for 2.0

**Comment by koskimas** [comment]
As a workaround, you can use the query context

```js
class Person extends Model {
  modifiers: {
    filters: query => query.where({ age: query.context().age })
  }
}

Person.query()
  .mergeContext({ age: 21 })
  .modify('filters')
```

**Comment by koskimas** [comment]
Fixed in v2.0 branch.
